mysql-test/suite/innodb/t/foreign_key.test

```diff
@@ -1186,6 +1186,29 @@ DROP TABLE binaries, collections;
 
 --echo # End of 10.6 tests
 
+CREATE TABLE t1
+(
+f1 VARCHAR(32)BINARY NOT NULL,
+f2 VARCHAR(32)BINARY NOT NULL,
+PRIMARY KEY (f1),
+INDEX(f2)
+) ENGINE=InnoDB;
+INSERT INTO t1 VALUES("MySQL", "InnoDB"), ("MariaDB", "NDB");
+
+CREATE TABLE t2
+(
+f1 VARCHAR(32)BINARY NOT NULL,
+f2 VARCHAR(255)BINARY NOT NULL,
+f3 int, PRIMARY KEY (f1), INDEX(f1), INDEX(f2)
+) ENGINE=InnoDB;
+INSERT INTO t2 VALUES("MySQL", "MySQL", 1),
+                     ("MariaDB", "MariaDB", 1);
```

Comment:
In the SQL standard, double quotes `"` are for quoting identifiers (such as table or column names that contain spaces), not for string literals. Let’s try to use the standard syntax where possible: `'MariaDB'`, not `"MariaDB"`.